diff --git a/sources/HOS/initial_condition.f90 b/sources/HOS/initial_condition.f90
index 701f9d5..7a2a884 100755
--- a/sources/HOS/initial_condition.f90
+++ b/sources/HOS/initial_condition.f90
@@ -4071,7 +4071,7 @@ READ(unit,1003) A1,time_i,A2,m_i,A3,n_i
 IF(m_i == m1 .and. n_i == n2) THEN
     DO i2 =  1, n2
         DO i1 = 1, m1
-            READ(unit,1004) eta(i1,i2),phis(i1,i2)
+            READ(unit,*) eta(i1,i2),phis(i1,i2)
         ENDDO
     ENDDO
     eta = eta / L_out
diff --git a/sources/HOS/output.f90 b/sources/HOS/output.f90
index 37ed38a..c10d7f1 100755
--- a/sources/HOS/output.f90
+++ b/sources/HOS/output.f90
@@ -760,9 +760,9 @@ IF (i_3D == 1) THEN
         ENDDO
     ENDIF
     !
-   102 FORMAT(3(ES12.5,X),ES12.5)
-   103 FORMAT(A,ES12.5,A,I5,A,I5)
-   104 FORMAT((ES12.5,X),ES12.5)
+   102 FORMAT(3(ES25.16E3,X),ES25.16E3)
+   103 FORMAT(A,ES25.16E3,A,I5,A,I5)
+   104 FORMAT((ES25.16E3,X),ES25.16E3)
    105 FORMAT(5(ES12.5,X),ES12.5,X)
    !
 ELSEIF (i_3D == 2) THEN
